src/e3/testsuite/report/rewriting.py

```diff
@@ -0,0 +1,243 @@
+                serialized_result = result_fp.read().strip()
+                if len(serialized_result) == 1:
+                    status = {
+                        "O": "OK",
+                        "I": "FAIL",
+                        "X": "XFAIL",
+                        "U": "UOK",
+                        "T": "VERIFY",
+                        "N": "DEAD",
+                        "n": "NOT-APPLICABLE",
+                        "C": "PROBLEM",  # Could also be CRASH/TIMEOUT
+                        "D": "DIFF",
+                    }[serialized_result]
+                else:
+                    status = serialized_result.split(":", 1)[0]
+
+            # Consider that there is a diff if we get a DIFF ("D")
+            if status == "DIFF":
+                # Fetch the new baseline: look for an X.out file in the same
+                # directory as "filename" (X.result). If not found, consider
+                # that the new baseline is empty.
+                new_baseline = b""
+                out_file = os.path.join(
+                    os.path.dirname(filename), f"{test_name}.out"
+                )
+                if os.path.exists(out_file):
+                    with open(out_file, "rb") as baseline_fp:
+                        new_baseline = baseline_fp.read()
+
+                self.handle_test_diff(
+                    summary, test_name, new_baseline, self.default_encoding
+                )
+
+            # Consider that the test aborted because of a fatal error if we get
+            # a CRASH/PROBLEM/TIMEOUT ('C') or a simple FAIL.
+            elif status == "PROBLEM":
+                self.handle_test_error(
+                    summary, test_name, "test status is CRASH/PROBLEM/TIMEOUT"
+                )
+
+    def handle_test_error(self,
+                          summary: RewritingSummary,
+                          test_name: str,
+                          reason: str) -> None:
+        """Notify users that a test result is an error."""
+        summary.errors.add(test_name)
+        self.print_warning(f"cannot update baseline for {test_name}: {reason}")
+
+    def handle_test_diff(self,
+                         summary: RewritingSummary,
+                         test_name: str,
+                         new_baseline: bytes | str,
+                         encoding: str) -> None:
+        """Rewrite the baseline of a single test."""
+        filename = self.baseline_filename(test_name)
+        baseline_exists = os.path.exists(filename)
+
+        baseline_bytes = self.postprocess_baseline(
+            new_baseline
+            if isinstance(new_baseline, bytes)
+            else new_baseline.encode(encoding)
+        )
+        if baseline_bytes:
+            if not baseline_exists:
+                self.print_info(
+                    f"no baseline file for {test_name}, creating it"
+                )
+            summary.new_baselines.add(test_name)
```

Comment:
This line should be in the `if` above @pmderodat. Sorry for having missed that.

Reply:
Ah thanks! Rewriting summaries are actually a blind spot of the tests I’ve written. I’ll take care of fixing this and adding necessary tests, thanks!